code/model/modelread.cpp

```diff
@@ -6079,6 +6086,13 @@ uint align_bsp_data(ubyte* bsp_in, ubyte* bsp_out, uint bsp_size)
 			memcpy(&bsp_chunk_size, bsp_in + 4, 4);
 		}
 
+		//Chunk size validation, if fails change it to copy the remaining data in chain
+		auto max_size = end - bsp_in;
+		if (bsp_chunk_size < 0 || bsp_chunk_size > max_size) {
```

Comment:
Methinks neither the change to `int` nor the check for <0 is needed.  A negative number, interpreted as an unsigned value, will always be larger than the largest possible signed value.

Reply:
you were right about that, it didnt even crossed my mind at the time. I changed it to use uint now.